proxy/src/schema/registry.rs

```diff
@@ -118,11 +125,22 @@ impl Registry {
             .await?
             .result?;
 
+        let register_metadata_vec = if let Some(pid_string) = maybe_pid {
+            let pid_cbor = ProjectIdCbor(pid_string);
+            to_vec(&pid_cbor)?
```

Comment:
What is the rationale for going for a vec?

Given our requirements it won't help much with backward and forward compatibility as the order of the elements still neds to be known. I think a struct of sorts with fields in it will be safer for that. And we need to a have a way to know which version the metadata we face was serialisaed with.

Reply:
The vectorization is just how serde_cbor serializes information, casting objects to vectors of bytes so they can be cast back faithfully. Is there another way you think could be more appropriate?

I'll update the `ProjectIdCbor` to a `ProjectCbor` struct that can contain more information, like the metadata version.